### PR TITLE
Embracing the Product Mindset

### PR DESCRIPTION
**Pull request by zliu41**

### CODE REVIEW

**Comment by colll78** [comment]
If anyone wants to spend their three working days in the sprint working with Plutarch on a smart contract protocol that is pushing the absolute limits of PlutusCore and practical design patterns on Cardano, than please feel free to reach out. 

The project is Midgard, an optimistic rollup scaling solution where the state of the layer 2 lives in a smart contract on the Cardano L1, and a very, very large number of validators collectively ensure the correctness of the transitions of the L2 state data.

**Review by effectfully — APPROVED** [review]
Looks great to me.

File: doc/product-mindset.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- Increased innovation
+- Improved empathy for users
+- Gaining broader knowledge and new perspectives
+- Personal and professional growth
+
+## How
+
+We'll introduce an additional sprint activity, alongside the two we currently have (making a release, and triaging issues and CIPs).
+
+Every sprint, an engineer is chosen, who should allocate up to 3 working days on a task in one of the following categories of their choice:
```
Comment (kwxm) [inline doc/product-mindset.md:23]: We'll have to be careful not to be too ambitious with the tasks that we take on.  There are plenty of things that we might try to do that could turn out to require a lot more than three days.

**Review by kwxm — APPROVED** [review]
This seems like a very good idea.  There are lots of things going on that it would be good to keep up with, but we've never been able to spare the time to do that.  Making it an official task will make that a lot easier.  

I can see this plan evolving as time goes on, but this seems like an excellent starting point.

**Review by ana-pantilie — APPROVED** [review]
I'm really happy we will get the time to do this, I think it's very valuable. I agree with Kenneth's points.

**Comment by Unisay** [comment]
I appreciate the goal of fostering a deeper understanding of the product domain among our engineers.  Personally I'd be happy to explore the ecosystem and learn from it.

However, I'd also like to highlight the importance of not overdoing this activity, the same way it doesn't make much sense for a product role to go and invest more time learning engineering, except maybe some low hanging low investment opportunities which deliver useful high-level insights / learnings.

**Comment by colll78** [comment]
> I appreciate the goal of fostering a deeper understanding of the product domain among our engineers. Personally I'd be happy to explore the ecosystem and learn from it.
> 
> However, I'd also like to highlight the importance of not overdoing this activity, the same way it doesn't make much sense for a product role to go and invest more time learning engineering, except maybe some low hanging low investment opportunities which deliver useful high-level insights / learnings.

Regardless of whether anyone decides to spend the time on Midgard or not, I think it would be helpful if during those three days you could pair-program with an experienced smart contract developer. Many of these developers would benefit greatly from the plutus core team being empathetic to the pains of production smart contract development on Cardano, and as such have a big incentive to offer to pair (myself included). I can recommend a number of such developers who would be willing to do such a pairing, and personally I would be happy to.

**Pull request merged**